### DISCUSSION TITLE
DualShock3 sometimes fail to connect in the latest version, but worked well in older versions.

### DISCUSSION THREAD

**Discussion by lkchx123** [nefarius/BthPS3#65] [Q&A]
First of all thanks to the author for making this awesome driver.

I updated the new version BthPS3 (v2.6.174), which fixed BSODs I encountered before, but now there is only a small chance that the handle can be connected to the computer via Bluetooth. Most of the time when I press the PS button, the computer recognizes it, shows it in the device manager and in the device list of DsHidMini, but the four LEDs of the handle are still blinking, and the keys don't work, and it's not actually connected to the computer. But it connects fine occasionally. Wired connection is normal.I've uninstalled and reinstalled multiple times and have the same problem. 

![image](https://user-images.githubusercontent.com/46598881/208323452-561191d6-b4cb-4530-b294-9b8bec3a6d7a.png)
Then I tried to uninstall and install the old version of BthPS3, without encountering the above situation, everything works fine. Older versions may have a version number of v2.2.282.0.
![image](https://user-images.githubusercontent.com/46598881/208323846-eeeec62b-2728-460d-90e8-e3207eba0d48.png)

May I ask what may be the reason? grateful!

---
中文：
首先感谢作者做出这个很棒的驱动。

我更新了新版本BthPS3(v2.6.174)，解决了之前遇到的蓝屏问题，但是现在手柄只有很小的几率能蓝牙连接到电脑。大部分时候当我按下PS按钮时，电脑能识别到它，显示在设备管理器和DsHidMini的设备列表中，但手柄的四个LED仍然在闪烁，按键也无效，实际上没有连接到电脑。但是它偶尔可以正常连接。有线连接是正常的。我多次卸载并重新安装，也遇到一样的问题。

然后我又尝试卸载并安装了旧版本的BthPS3，没有遇到上述情况，一切正常工作。旧版本的版本号可能是v2.2.282.0。

请问可能是什么原因？感谢！

**Comment by nefarius** [comment]
Please [make a trace](https://vigem.org/projects/BthPS3/Debugging-the-drivers/) when the error happens and share the results here.

**Reply by lkchx123** [reply]
Thanks for your reply. Here is the BthPS3.etl file.
https://drive.google.com/file/d/10NkoVQ5KWGUbYQdKICa4bcgeW5n4_vEK/view?usp=sharing

I did the connect operation three times. When I press the PS button for the first time, the computer does not respond. The second and third time, the gamepad is displayed in the device list, but the led is still blinking and then off. And it is still in the device list after the led is off.

**Reply by nefarius** [reply]
I don't see any errors in that trace so it looks like the controller simply initiates the disconnect.

**Reply by lkchx123** [reply]
All right, thank you anyway. I have to go back to the old version.
